### DISCUSSION TITLE
CSV file transfer

### DISCUSSION THREAD

**Discussion by ywan614** [msiemens/tinydb#380] [Q&A]
Is there any module can convert CSV or SQL file to TinyDB json file?

**Comment by therden** [comment]
Hello @ywan614.

I've created [a small Github repo](https://github.com/therden/convert_csv_to_TinyDB) containing two small files of Python code and a three-record text file to demonstrate two ways to import data from a `csv` file into **TinyDB**.  Which approach you use will depend on 

a) whether the source records contain some type of `record ID` field and, if so, 
b) how you want to handle that during the conversion.

Each of these programs assumes that the first line of the file contains the comma-separated column/field names for the data, and after the conversion each pretty-prints the contents of the **TinyDB** file to the screen.

Hopefully they will be enough to get you started.

**Comment by msiemens** [comment]
Converting a CSV file should be possible by using the [`csv.DictReader`](https://docs.python.org/3/library/csv.html#csv.DictReader) and inserting the resulting CSV record dicts into TinyDB.

> SQL file

That's gonna be more complicated. SQL files are meant to be understood by the specific SQL engines used by the respective databases. MySQL's SQL files may use features not supported by SQLite and vice versa. It would be possible to parse a subset of SQL (e.g. [python-sqlparse](https://sqlparse.readthedocs.io/en/latest/)) to read into TinyDB but I don't know of any modules that would accomplish that.
